src/routes/controllers/user.controller.js

```diff
@@ -0,0 +1,33 @@
+const User = require("../../../models/User");
+
+const logger = require("../../../libs/logger");
+
+const findOrCreateUser = async (req, res, next) => {
+  try {
+    const { email, displayName } = req.body;
+    const profileUrl = req.body.profile_url;
+
+    const existUser = await User.findOne({ email }).lean().exec();
```

Comment:
위 로직을 수행하기 전에 `email` 값이 `null`인지 체크하는 로직을 넣으면 어떨까요?
(물론 스키마에서 필수값으로 입력해야 한다고 되어있기 때문에, `null`이라면 `create`할 때 에러가 발생할 것으로 예상돼서  꼭 필요한 건 아닙니다!)

```javascript
if (!email) {
  const error = new Error("이메일은 필수값입니다")
  logger.error(error.toString());
}

Reply:
근데 만약에 쓴다면, `next(error)`가 낫겠군요!

Reply:
```suggestion
if (!email) {
  const error = createError(400, "Invalid input: email");
  logger.error(error.toString());

  throw error;
}

```
반영해봤는데 이렇게 작성하면 어떨까요?? http-error 라이브러리에서 상태코드를 넣으면 앞에 에러 타입을 자동으로 붙혀주네요! 송이님 덕에 좋은 거 배웠습니다 👍  아래는 에러 로그에요!

`2022-10-17 16:19:39.971[error] BadRequestError: Invalid input: email`

Reply:
오 `createError`로 작성하는 걸 놓쳤는데, 감사합니다!
`next(error)`로 보내서 에러는 에러 핸들러에서 모두 처리해요:)